src/format/parse.rs

```diff
@@ -1040,3 +1298,135 @@ fn test_issue_1010() {
+    // ambiguous, interpreted as calendar date
+    assert!(parse("+02023160 ").is_err());
+    assert!(parse("+12023160 ").is_err());
+    // No padding
+    assert_eq!(parse("2023-16 "), Err(INVALID));
+    assert_eq!(parse("2023-1 "), Err(INVALID));
+    assert_eq!(parse("23-160 "), Err(INVALID));
+
+    let from_isoywd_opt = NaiveDate::from_isoywd_opt;
+    // week date, basic format
+    assert_eq!(parse("2023W235 "), Ok((from_isoywd_opt(2023, 23, Fri).unwrap(), " ")));
+    // week date, extended format
+    assert_eq!(parse("2023-W23-5 "), Ok((from_isoywd_opt(2023, 23, Fri).unwrap(), " ")));
+    // week date, basic format, expanded representation
+    assert_eq!(parse("-2023W235 "), Ok((from_isoywd_opt(-2023, 23, Fri).unwrap(), " ")));
+    assert_eq!(parse("+2023W235 "), Ok((from_isoywd_opt(2023, 23, Fri).unwrap(), " ")));
+    assert_eq!(parse("+02023W235 "), Ok((from_isoywd_opt(2023, 23, Fri).unwrap(), " ")));
+    assert_eq!(parse("+12023W235 "), Ok((from_isoywd_opt(12023, 23, Fri).unwrap(), " ")));
+    // calendar date, extended format, expanded representation
+    assert_eq!(parse("-2023-W23-5 "), Ok((from_isoywd_opt(-2023, 23, Fri).unwrap(), " ")));
+    assert_eq!(parse("+2023-W23-5 "), Ok((from_isoywd_opt(2023, 23, Fri).unwrap(), " ")));
+    assert_eq!(parse("+02023-W23-5 "), Ok((from_isoywd_opt(2023, 23, Fri).unwrap(), " ")));
+    assert_eq!(parse("+12023-W23-5 "), Ok((from_isoywd_opt(12023, 23, Fri).unwrap(), " ")));
+    // mixed basic and extended format
+    assert_eq!(parse("2023-W235 "), Err(INVALID));
+    assert_eq!(parse("202306-W235 "), Err(INVALID));
+    assert_eq!(parse("-2023-W235 "), Err(INVALID));
+    // No padding
+    assert_eq!(parse("2023-W25 "), Err(INVALID));
+    assert_eq!(parse("23-W23-5 "), Err(INVALID));
+    // Year is interpreted as `iso_year`
+    assert_eq!(parse("2022-W52-7 "), Ok((NaiveDate::from_ymd_opt(2023, 1, 1).unwrap(), " ")));
+}
+
+#[cfg(test)]
+#[test]
+fn test_parse_iso8601_time() {
+    fn parse(s: &str) -> ParseResult<(&str, u32, u32, u32, u32, bool)> {
+        let mut parsed = Parsed::new();
+        let (s, _, hour24) = parse_iso8601_time(&mut parsed, s)?;
+        Ok((
+            s,
+            12 * parsed.hour_div_12.unwrap() + parsed.hour_mod_12.unwrap(),
+            parsed.minute.unwrap(),
+            parsed.second.unwrap_or(0),
+            parsed.nanosecond.unwrap_or(0),
+            hour24,
+        ))
+    }
+
+    // basic format, complete representation
+    assert_eq!(parse("152830 "), Ok((" ", 15, 28, 30, 0, false)));
+    // extended format, complete representation
+    assert_eq!(parse("15:28:30 "), Ok((" ", 15, 28, 30, 0, false)));
+    // basic format, fractional second
+    assert_eq!(parse("152830,6 "), Ok((" ", 15, 28, 30, 600_000_000, false)));
+    assert_eq!(parse("152830.60 "), Ok((" ", 15, 28, 30, 600_000_000, false)));
+    // extended format, fractional second
```

Comment:
What about bad fractional char?

     assert_eq!(parse("152830|60 "), Err(INVALID));

Reply:
It would just be valid, similar to how `15:28:30 ` stops parsing at ` ` it would here stop parsing the time at `|`.